src/Solvers/Greedy.jl

```diff
@@ -172,6 +178,55 @@ function greedy_choose_simple!(queue, remaining)
     return node
 end
 
+
+function greedy_choose_thermal!(queue, remaining, nbranch=8, temperature=1, rel_temperature=true)
+    n = 0
+    choices = Vector{HeapNode}()
+
+    while (!isempty(queue) && n < nbranch)
+        node = pop!(queue)
+        a, b, c = meta(node)
+
+        any(inds -> inds ∉ values(remaining), (a, b)) && continue
+
+        push!(choices, node)
+        n += 1
+    end
+
+    if n == 0
+        return nothing
+    elseif n == 1
+        return choices[1]
+    end
+
+    costs = [choice.cost for choice in choices]
+    cmin = costs[1]
+
+    rel_temperature && (temperature *= max(1, abs(cmin)))
+
+    if temperature == 0.
+        energies = [c == cmin ? 1 : 0 for c in costs]
+    else
+        energies = [exp(-(c - cmin) / temperature) for c in costs]
+    end
```

Comment:
This code is type unstable. If `temperature` is approximately `0.`, then `energies` is a `Vector{Int}`. Otherwise, `energies` is a `Vector{Float64}`. Use `Float32` on both (`Float64` is probably too much).

Also the code can be refactored in the following way to be much cleaner:

```suggestion
    energies = if temperature == 0.
        [c == cmin ? 1. : 0. for c in costs]
    else
        [exp(-(c - cmin) / temperature) for c in costs]
    end
```